### ISSUE TITLE
spotbugs report: broken links with inner classes

### ISSUE THREAD

**Issue by davewichers** [spotbugs/spotbugs-maven-plugin#236]
I believe this is basically the same issue another tool had as documented here: https://issues.apache.org/jira/browse/SUREFIRE-1443

In my project, I noticed warnings like:
[warn] [XHTML Sink] Modified invalid anchor name: 'org.owasp.esapi.SecurityConfiguration$Threshold' to 'org.owasp.esapi.SecurityConfigurationThreshold'

But when I looked in the generated spotbugs.html file, I noticed the error on this line that was generated:
`<h3><a name="org.owasp.esapi.SecurityConfiguration.24Threshold"></a>org.owasp.esapi.SecurityConfiguration$Threshold</h3>`

Notice the .24 instead of the $ in the value of the name property. If it generated this:
`<h3><a name="org.owasp.esapi.SecurityConfiguration$Threshold"></a>org.owasp.esapi.SecurityConfiguration$Threshold</h3>`

Then the link to the anchor tag works. To fix this I think you simply need to decode the full classname before putting it's value in the name property. Not sure if that would get rid of this warning message or if that is something else. But this seems to fix the bug in the .html.

I'm using the 4.0.4 version of the spotbugs-maven-plugin to generate the report.

**Comment by davewichers** [comment]
I have no idea how to correct this. I just noticed the bug and describe what the correct output should be. I also noticed related discussions on the same problem in other tools, and @KengoTODA decided this issue belongs to this project rather than SpotBugs itself. Basically, there is a step missing somewhere where a String needs to be decoded before being included in the output, as an encoded value of .24 is being output rather than the original $ symbol.

**Comment by davewichers** [comment]
Hey - its been 8 months. Any progress on this? I happened to notice it again today :-)

**Comment by hazendaz** [comment]
Please reconfirm on latest and if possible a PR.  This has not been looked at outside of notes here.

Sent from my Verizon, Samsung Galaxy smartphone
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by davewichers** [comment]
Pull Request? For what? I don't know how to fix this issue. I can confirm I'm using: Spotbugs-maven-plugin 4.6.0.0 and I still have this problem.

**Comment by hazendaz** [comment]
@davewichers Is this sill an issue?

**Comment by davewichers** [comment]
Yes - Unless you've fixed it. Do you think you fixed it?

**Comment by hazendaz** [comment]
Hi @davewichers, likely not fixed, was just checking due to time lapse here.  I'm more or less now ready to release.  Will take another look at this to see if I can resolve before release.  Thanks.

**Comment by hazendaz** [comment]
@davewichers Resolved this.  Issue was a bit different than original posted here in that the 24 still remains and is fine.  The actual issue was that $ was removed line above it which prevented click.

Using esapi from roughly point in time you reported this from commit b6f8808e, I was able to produce this data area using existing release.  The latest esapi isn't running this report at all so that was reason to drop back to that point to see the warnings and resolve it.

```
<td>Medium</td></tr></table></section><a name="org.owasp.esapi.SecurityConfigurationThreshold"></a><section>
<h3><a name="org.owasp.esapi.SecurityConfiguration.24Threshold"></a>org.owasp.esapi.SecurityConfiguration$Threshold</h3>
```

Note line 1 there, its missing $ between SecurityConfiguration and Threshold.

After applying same fix that surefire did and dealing with how groovy handles closures, I was able to get this result.

```
<td>Medium</td></tr></table></section><a name="org.owasp.esapi.SecurityConfiguration$Threshold"></a><section>
<h3><a name="org.owasp.esapi.SecurityConfiguration.24Threshold"></a>org.owasp.esapi.SecurityConfiguration$Threshold</h3>
```
Note 24 still exists but the first line has $ now.  That $ fix is all that surefire was doing.  Now the click works for that one.
